packages/evm/src/evm.ts

```diff
@@ -137,7 +138,23 @@ export class EVM implements EVMInterface {
 
   protected readonly _emit: (topic: string, data: any) => Promise<void>
 
-  constructor(opts: EVMOpts = {}) {
+  private bn128: {
+    ec_pairing: (input_str: string) => string
+    ec_add: (input_str: string) => string
+    ec_mul: (input_hex: string) => string
+  }
+
+  static async create(opts: EVMOpts) {
+    if (opts.bn128 === undefined) {
+      opts.bn128 = await initRustBN()
```

Comment:
I thought about that but this isn't really "custom crypto".  This is just our existing dependency but since it now requires an async constructor, I'm having to pass it in.  In this case, I don't think it makes sense to instantiate it in Common or even attach it there. There isn't an alternative implementation of the crypto anywhere that I'm aware of aside from the one older ASM version we were previously using.